api/src/main/java/io/minio/MinioClient.java

```diff
@@ -913,32 +916,56 @@ private HttpResponse execute(Method method, String region, String bucketName, St
       request = Signer.signV4(request, region, accessKey, secretKey);
     }
 
-    if (this.traceStream != null) {
-      this.traceStream.println("---------START-HTTP---------");
-      String encodedPath = request.httpUrl().encodedPath();
-      String encodedQuery = request.httpUrl().encodedQuery();
-      if (encodedQuery != null) {
-        encodedPath += "?" + encodedQuery;
+    Response response = null;
+
+    // Loop to retry with new location of 307.
+    for (int i = 0; i < TEMP_REDIRECT_TRY_COUNT; i++) {
```

Comment:
The loop is continued only for 307.  What is all the process in the loop?

My suggestion to 307 is that SDK should error out instead of retrying its own.  Auto-retry doesn't solve real problem ie put object doesn't work (not only for minio-java).

Minio server returns 307 only for `http` to `https`.  `https` would invite other problems like ignoring self signed certificate.